### PR TITLE
[BEAM-5092]  Optimize Row comparison. 

### PR DESCRIPTION
**Pull request by reuvenlax**

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoder.java [hunk 1]
```diff
@@ -72,6 +75,12 @@ public static RowCoder of(Schema schema) {
   }
```
Comment (timrobertson100) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoder.java]: Should this assign a random ID if the provided Schema is already with a UUID?

Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoder.java]: You are correct - this was the cause of the test failures. Fixed now.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoder.java [hunk 2]
```diff
@@ -72,6 +75,12 @@ public static RowCoder of(Schema schema) {
...
 
   private RowCoder(Schema schema, UUID id) {
+    if (schema.getUUID() != null) {
+      checkArgument(schema.getUUID().equals(id));
```
[the remark this answers is not in the document]
Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoder.java]: Done.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoder.java [hunk 3]
```diff
@@ -71,12 +71,17 @@
...
+    UUID id = (schema.getUUID() == null) ? UUID.randomUUID() : schema.getUUID();
+    return new RowCoder(schema, id);
   }
 
   private RowCoder(Schema schema, UUID id) {
     if (schema.getUUID() != null) {
-      checkArgument(schema.getUUID().equals(id));
+      checkArgument(
+          schema.getUUID().equals(id),
+          "Schema has a UUID that doesn't match argument to constructor. %s v.s. $s",
```
Comment (timrobertson100) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoder.java]: "Preconditions only accepts the %s placeholder in error message strings"

2 minor typos (one fails build)

"Schema has a UUID that doesn't match argument to constructor. %s vs. %s",

**Comment by reuvenlax** [comment]
retest this please

**Comment by reuvenlax** [comment]
retest this please

**Comment by reuvenlax** [comment]
retest this pelase

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
@timrobertson100  all comments addressed

**Pull request merged**